### PR TITLE
Change MembershipListener, Remove InitialMembershipListener

### PR DESCRIPTION
**Pull request by yemreinci**
Inherits the similar API pattern that was already adopted for other listener types. (see: #604 #607 #614)

* `MembershipListener` is now a concrete type on which callable event handlers are registered.
* `InitialMembershipListener` is removed, its functionality is available via `MembershipListener::on_init`.

### CODE REVIEW

**Comment by yemreinci** [comment]
verify-linux

**Comment by yemreinci** [comment]
verify-linux

**Comment by yemreinci** [comment]
verify-linux

File: examples/monitoring/cluster/main.cpp [hunk 1]
```diff
@@ -14,77 +14,72 @@
...
-        std::cout << "[MyMemberListener::memberAdded] New member joined:" << membershipEvent.getMember().getAddress() <<
-        std::endl;
-    }
+MembershipListener makeMembershipListener() {
+    return MembershipListener()
+        .on_added([](const hazelcast::client::MembershipEvent &membershipEvent) {
+            std::cout << "[MyInitialMemberListener::memberAdded] New member joined:" <<
+            membershipEvent.getMember().getAddress() << std::endl;
+        })
+        .on_removed([](const hazelcast::client::MembershipEvent &membershipEvent) {
```
Comment (ihsandemir) [inline examples/monitoring/cluster/main.cpp:28]: should we better name these as `on_joined`, `on_left`? meaning that member joined to cluster, member left from cluster.

**Comment by yemreinci** [comment]
verify-linux

**Review by ihsandemir — APPROVED** [review]

**Comment by yemreinci** [comment]
verify-linux

**Comment by yemreinci** [comment]
verify-linux

**Comment by yemreinci** [comment]
verify-linux

**Pull request merged**